### ISSUE TITLE
'cmd_frame_id' vs 'this->cmd_frame_id' in pf_local_planner.cpp?

### ISSUE THREAD

**Issue by sayali-purdue** [CARMinesDouai/MultiRobotExplorationPackages#1]
The line #360 in 'adaptive_local_planner/src/pf_local_planner.cpp' is looking for a transform between the frames 'cmd_frame_id' and 'this->cmd_frame_id' (shown below):

`this->tf->lookupTransform(cmd_frame_id, this->cmd_frame_id, ros::Time(0), transform);`

The source frame 'this->cmd_frame_id' is set to 'base_link' as per the code, but I am not able to figure out the target frame 'cmd_frame_id'.

Which frame is expected for target frame 'cmd_frame_id'? 

Would really appreciate your help in understanding this transform lookup. Thank you!

**Comment by lxsang** [comment]
They are the same variables I think, it is really weird this transformation but the general idea is to get the robot position and orientation from the base_link frame.

**Comment by sayali-purdue** [comment]
Thank you for the response.

If these variables are same, this lookup is between the same source and target frames. This is an identity transform. Then, can you please explain why is there a need for the function `my_pose`? Am I missing some detail? 

Also, can you please clarify which part of the robot is referenced here in the term "robot position and orientation". In particular, which robot frame is the target frame, when 'base_link' is the source frame?

Thanks again!

**Comment by lxsang** [comment]
Performing this identity transform is just to get the transform object (instance of `StampedTransform`), then we get the robot pose on the `base_link` frame using the `getOrigin`  method and robot orientation using the `getRotation` method from that object. Note that, what we want here is the robot pose and orientation relative to the `base_link` frame, not the global pose. 

There is maybe another way to implement this without causing confusion.

**Comment by sayali-purdue** [comment]
Thanks again.

I am, now, trying to understand:
1. why lookupTransform instead of just initializing an identity transform object?
2. why is identity transform object required? The origin and rotation of this object would be (0,0,0) and (0,0,0,1) respectively. I see that there are a number of computations that use the origin and rotation of this object. But, if those values are going to be zero, then are those values really required to be included in the computation of distance between the robot and the goal. There is already a lookup for the transform from goal_frame_id to base_link, which can be used directly to get the distance of the goal from the robot.

Would really appreciate if you could help me to understand the above points.
Thanks

**Comment by lxsang** [comment]
You are right, we don't need the my_pose function at all since it always returns an identity transform. This function does not effect the correctness of the algorithm, but it is completely unnecessary and can cause confusion.

I will clean up the code by remove the use of this function. We can continue this discussion on my github: [https://github.com/lxsang/ROS-packages](https://github.com/lxsang/ROS-packages), il will make the changes there.
